Terminal.Gui/Views/ComboBox.cs

```diff
@@ -16,7 +16,7 @@ public class ComboBox : View
     private readonly ComboListView _listview;
     private readonly int _minimumHeight = 2;
     private readonly TextField _search;
-    private readonly IList _searchset = new List<object> ();
+    private readonly ObservableCollection<object> _searchSet = [];
```

Comment:
> A type like `IEnumerable<T>` or `IList<T>` is much more friendly.

But none of them also raise a collection change notification.

Reply:
I know. You have to do it manually.

But that's the consumer's responsibility - not ours.

If _we_ need to react to collection changes, then the collection shouldn't even be exposed to the consumer, because they can circumvent it.

Reply:
Ultimately, the key is that second sentence - it's their responsibility.

We don't know, ahead of time, when or if they want things to happen.

What if they want to make several changes to the collection before redrawing?

What if they don't want to re-draw at all?

What if they want to re-draw when a property of something _in_ the collection changes?

There are pretty much infinite possibilities, so we shouldn't try to end-run the consumer.

Reply:
And that makes it easier on us, anyway. :)

Reply:
I am aware of your comments and subscribe. But as I already explained, I just wanted to provide an event for when a source collection is changed so that the user can take the appropriate actions. If the source has to be manipulated beforehand and the user doesn't want to be bombarded with a bunch of events, he can first set the source to null and manipulate the data he wants and then go back to defining the already changed source.
Searching on Google about the best way to have a collection that notifies the client about changes, what they most advise to use is the `ObservableCollection<T>` or create a new one. If one already exists that works well, the user can create a collection that is compatible with IList<T> and only when defining the Source property must it be created as `new ListWrapper<string> (new ObservableCollection<string> (source))`.